ttnn/cpp/ttnn/operations/moreh/moreh_cumsum/device/moreh_cumsum_device_operation.cpp

```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_cumsum_device_operation.hpp"
+
+#include <cstdint>
+
+#include "tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "ttnn/operation.hpp"
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::moreh::moreh_cumsum {
+void MorehCumsumDeviceOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    const auto dim = operation_attributes.dim;
+    TT_ASSERT((dim >= 0 && dim <= 3), "dim should be 0 - 3");
```

Comment:
TT_ASSERT is removed in release builds..
Please use TT_FATAL and include the value of the dim in the message for easier debugging.

Reply:
I've updated the code to use TT_FATAL and include the dimension value in the message for better debugging